c_src/membrane_element_ffmpeg_swresample/converter.c

```diff
@@ -1,138 +1,135 @@
-    default:
-      ret_val = -1;
+  if (membrane_sample_fmt_to_av_sample_fmt(src_format, 0, &from_s24le,
+                                           &src_av_format)) {
+    err_reason = "unsupported_src_format";
+    goto error;
   }
-  return ret_val;
-}
-
-static int nb_channels_to_av_layout(int channels, int64_t* av_layout) {
-  int ret_val = 0;
-  switch (channels) {
-    case 1: *av_layout = AV_CH_LAYOUT_MONO; break;
-    case 2: *av_layout = AV_CH_LAYOUT_STEREO; break;
-    default:
-      ret_val = -1;
+  if (membrane_sample_fmt_to_av_sample_fmt(dst_format, 1, NULL,
+                                           &dst_av_format)) {
+    err_reason = "unsupported_dst_format";
+    goto error;
   }
-  return ret_val;
-}
-
-static ERL_NIF_TERM export_create(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[]) {
-  UNUSED(argc);
-  MEMBRANE_UTIL_PARSE_UINT_ARG(0, src_format);
-  MEMBRANE_UTIL_PARSE_UINT_ARG(1, src_rate);
-  MEMBRANE_UTIL_PARSE_UINT_ARG(2, src_channels);
-  MEMBRANE_UTIL_PARSE_UINT_ARG(3, dst_format);
-  MEMBRANE_UTIL_PARSE_UINT_ARG(4, dst_rate);
-  MEMBRANE_UTIL_PARSE_UINT_ARG(5, dst_channels);
-
-  enum AVSampleFormat src_av_format, dst_av_format;
-  int ret_val;
-  char from_s24le;
-  ret_val = membrane_sample_fmt_to_av_sample_fmt(src_format, 0, &from_s24le, &src_av_format);
-  if(ret_val) return membrane_util_make_error_args(env, "src_format", "Unsupported sample format");
-  ret_val = membrane_sample_fmt_to_av_sample_fmt(dst_format, 1, NULL, &dst_av_format);
-  if(ret_val) return  membrane_util_make_error_args(env, "dst_format", "Unsupported sample format");
   int64_t src_layout, dst_layout;
-  ret_val = nb_channels_to_av_layout(src_channels, &src_layout);
-  if(ret_val) return membrane_util_make_error_args(env, "src_channels", "Unsupported number of channels");
-  ret_val = nb_channels_to_av_layout(dst_channels, &dst_layout);
-  if(ret_val) return membrane_util_make_error_args(env, "dst_channels", "Unsupported number of channels");
-
-  ConverterHandle *handle = enif_alloc_resource(RES_CONVERTER_HANDLE_TYPE, sizeof(ConverterHandle));
-  char* init_error = init(
-    handle,
-    from_s24le,
-    src_av_format, src_rate, src_layout,
-    dst_av_format, dst_rate, dst_layout
-  );
-  if(init_error)
-    return membrane_util_make_error_internal(env, init_error);
+  if (nb_channels_to_av_layout(src_channels, &src_layout)) {
+    err_reason = "unsupported_src_no_channels";
+    goto error;
+  }
+  if (nb_channels_to_av_layout(dst_channels, &dst_layout)) {
+    err_reason = "unsupported_dst_no_channels";
+    goto error;
+  }
 
-  ERL_NIF_TERM converter_term = enif_make_resource(env, handle);
-  enif_release_resource(handle);
+  state = unifex_alloc_state(env);
+  err_reason = lib_init(state, from_s24le, src_av_format, src_rate, src_layout,
+                        dst_av_format, dst_rate, dst_layout);
+  if (err_reason) {
+    goto error;
+  }
 
-  return membrane_util_make_ok_tuple(env, converter_term);
+error:
```

Comment:
This label is confusing. Either rename it to fn_name_exit or place a happy path before the label and get rid of that if in the next line